eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -738,124 +845,144 @@ object Transactions {
+          lockTime = 0
+        )
+        ClaimLocalAnchorTx(fundingKey, commitKeys, input, tx, commitmentFormat)
+      })
     }
+  }
 
-    /**
-     *
-     * @param fundingKey       funding public keys
-     * @param commitKeys       commitment keys
-     * @param toLocal          true if this is the redeem info for the `toLocal` commit tx output
-     * @param commitmentFormat commitment format
-     * @return the redeem information for a local or remote commit tx anchor output
-     */
-    def redeemInfo(fundingKey: PublicKey, commitKeys: CommitmentPublicKeys, toLocal: Boolean, commitmentFormat: CommitmentFormat): RedeemInfo = {
-      commitmentFormat match {
+  /** This transaction claims our anchor output in a remote commitment. */
+  case class ClaimRemoteAnchorTx(fundingKey: PrivateKey, commitKeys: RemoteCommitmentKeys, input: InputInfo, tx: Transaction, commitmentFormat: CommitmentFormat) extends ClaimAnchorTx with RemoteCommitForceCloseTransaction {
+    override val desc: String = "remote-anchor"
+    override val redeemInfo: RedeemInfo = ClaimRemoteAnchorTx.redeemInfo(fundingKey.publicKey, commitKeys.publicKeys, commitmentFormat)
+
+    override def sign(): Transaction = sign(WalletInputs(Nil, None))
+
+    def sign(walletInputs: WalletInputs): Transaction = {
+      val toSign = copy(tx = setWalletInputs(walletInputs))
+      val witness = commitmentFormat match {
         case DefaultCommitmentFormat | _: AnchorOutputsCommitmentFormat =>
-          val redeemScript = Script.write(anchor(fundingKey))
-          RedeemInfo.P2wsh(redeemScript)
+          val redeemScript = Script.write(anchor(fundingKey.publicKey))
+          val sig = toSign.sign(fundingKey, sighash, RedeemInfo.P2wsh(redeemScript), walletInputs.spentUtxos)
+          witnessAnchor(sig, redeemScript)
         case _: SimpleTaprootChannelCommitmentFormat =>
-          val anchorKey = if (toLocal) commitKeys.localDelayedPaymentPublicKey.xOnly else commitKeys.remotePaymentPublicKey.xOnly
-          RedeemInfo.TaprootKeyPath(anchorKey, Some(Taproot.anchorScriptTree))
+          val Right(anchorKey) = commitKeys.ourPaymentKey
+          val redeemInfo = RedeemInfo.TaprootKeyPath(anchorKey.xOnlyPublicKey(), Some(Taproot.anchorScriptTree))
+          val sig = toSign.sign(anchorKey, sighash, redeemInfo, walletInputs.spentUtxos)
+          Script.witnessKeyPathPay2tr(sig)
       }
+      toSign.tx.updateWitness(toSign.inputIndex, witness)
     }
+  }
 
-    private def createUnsignedTx(input: InputInfo): ClaimAnchorOutputTx = {
-      val unsignedTx = Transaction(
-        version = 2,
-        txIn = TxIn(input.outPoint, ByteVector.empty, 0) :: Nil,
-        txOut = Nil, // anchor is only used to bump fees, the output will be added later depending on available inputs
-        lockTime = 0
-      )
-      ClaimAnchorOutputTx(input, unsignedTx)
+  object ClaimRemoteAnchorTx {
```

Comment:
nit:there is some redundancy between `ClaimLocalAnchorTx` and `ClaimRemoteAnchorTx` (sign, findInput, ...) that could be refactored in a common trait